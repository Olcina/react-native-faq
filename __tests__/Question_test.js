import React from 'react';
import { Question }  from '../index';

import renderer from 'react-test-renderer';

const action_example = () => {
    console.log("I'm an action!");
    
}

const bullets = [
    "I'm a bullet",
    "Second Bullet"
]

const question = {
    question: "Did I render?",
    reply: "It seems so:",
    bullets: bullets,
    actionText: "I'm ready to help!",
    onClick: action_example,
}

test('Question renders correctly', () => {
    const tree = renderer.create(
        <Question
            {...question}
        />).toJSON();
    expect(tree).toMatchSnapshot();
});
test('renders whitout action', () => {
    const tree = renderer.create(
        <Question
            question='test'
            reply='test'
        />).toJSON();
    expect(tree).toMatchSnapshot();
});