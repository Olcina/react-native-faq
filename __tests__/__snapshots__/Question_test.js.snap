// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Question renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "lightgrey",
        "borderRadius": 5,
        "margin": 2,
      },
    ]
  }
>
  <Text
    style={
      Array [
        Object {
          "fontSize": 20,
        },
      ]
    }
  >
    Did I render?
  </Text>
  <Text
    style={
      Array [
        Object {
          "fontSize": 15,
        },
      ]
    }
  >
    It seems so:
  </Text>
  <View
    style={
      Object {
        "display": "flex",
        "textAlign": "center",
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "marginLeft": "10%",
          },
        ]
      }
    >
      I'm a bullet
    </Text>
    <Text
      style={
        Array [
          Object {
            "marginLeft": "10%",
          },
        ]
      }
    >
      Second Bullet
    </Text>
  </View>
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "black",
        "borderRadius": 5,
        "margin": 5,
        "opacity": 1,
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "fontSize": 30,
            "textAlign": "center",
          },
        ]
      }
    >
      I'm ready to help!
    </Text>
  </View>
</View>
`;

exports[`renders whitout action 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "lightgrey",
        "borderRadius": 5,
        "margin": 2,
      },
    ]
  }
>
  <Text
    style={
      Array [
        Object {
          "fontSize": 20,
        },
      ]
    }
  >
    test
  </Text>
  <Text
    style={
      Array [
        Object {
          "fontSize": 15,
        },
      ]
    }
  >
    test
  </Text>
  <Text />
  <Text />
</View>
`;
